redwood-treehouse-guest/src/commonMain/kotlin/app/cash/redwood/treehouse/treehouseCompose.kt

```diff
@@ -69,7 +69,7 @@ private class RedwoodZiplineTreehouseUi(
 
     this.saveableStateRegistry = SaveableStateRegistry(
       restoredValues = stateSnapshot?.toValuesMap(),
-      canBeSaved = { true },
+      canBeSaved = { stateSnapshot?.canBeSaved() ?: true },
     )
```

Comment:
I'm not sure how `canBeSaved` can ever be `false`. We evaluate `stateSnapshot?.toValuesMap()` in `restoredValues` before the `canBeSaved` lambda is invoked. If this `toValuesMap()` throws an exception, we're not catching it anywhere.

Reply:
- `canBeSaved` is invoked during instantiation of `SaveableStateRegistryImpl`

- `canBeSaved` is a mechanism to signal `SaveableStateRegistryImpl` if `restoredValues` is serializable, it's in this case, the `canBeSaved` happen to come with the underlying implementation that utilizes `stateSnapshot?.toValuesMap()`

Reply:
> * `canBeSaved` is a mechanism to signal `SaveableStateRegistryImpl` if `restoredValues` is serializable

Maybe I'm misunderstanding something. Here are my assumptions:

* Calling `stateSnapshot?.toValuesMap()` will throw an exception if the values aren't serializable (I'm assuming this by the implementation of your `canBeSaved` function).
* When instantiating the `SaveableStateRegistry` instance, we're immediately invoking `stateSnapshot?.toValuesMap()` (on line 71).
* If that fails because `toValuesMap()` isn't serializable, then an exception is thrown.
* We never got to a point where the `canBeSaved` lambda is invoked, because we threw an exception while instantiating `SaveableStateRegistry`.

As an aside, would it be possible to add a unit test for this?

Reply:
yep, that's the right assumption. 

The point of this change is to serve as a reminder that `canBeSaved` needs to receive a fair share of attention.

and it's in this case, the `canBeSaved` happen to come with the underlying implementation that utilizes `stateSnapshot?.toValuesMap()`, this will not always be the case, thus the reminder.

unit test could make sense for `fromJsonElement` I think?